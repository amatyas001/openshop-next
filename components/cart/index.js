export * from './CartContent/CartContent';
export * from './CartAmount/CartAmount';
export * from './CartButton/CartButton';
export * from './CartItem/CartItem';
export * from './CartPanel/CartPanel';
