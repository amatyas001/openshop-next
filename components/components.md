Elements are composed by [Chakra-UI](https://chakra-ui.com) components and preserve the library features so [Style Props](https://chakra-ui.com/style-props) can be passed to them and accessibility remains optimized.

> _Style Props are a way to alter the style of a component by simply passing props to it. It helps to save time by providing helpful shorthand ways to style components._

For more information visit the [Chakra UI](https://chakra-ui.com) documentation.

Components are divided into two main category: _wrappers_ and _single responsibility_ elements. Wrapper components does not contain business logic, however there are special cases like _handling lifted states to keep sensitive data locally_ and not to populate them to the Redux store which is persisted on the machine.

**State Dependencies** and **Wrapped Components** are documented in each description to give brief information about the context of posible interactions of the actual component. _Examples_ prepared with proper state and may render extra controls to test different cases with ease.
