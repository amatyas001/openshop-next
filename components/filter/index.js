export * from './FilterButton/FilterButton';
export * from './FilterColor/FilterColor';
export * from './FilterPrice/FilterPrice';
export * from './FilterName/FilterName';
export * from './FilterPanel/FilterPanel';
export * from './FilterReset/FilterReset';
