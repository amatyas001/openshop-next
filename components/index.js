export * from './shared';
export * from './cart';
export * from './payment';
