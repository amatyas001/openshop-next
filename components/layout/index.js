export * from './Content/Content';
export * from './Footer/Footer';
export * from './ProductsTopBar/ProductsTopBar';
export * from './Navbar/Navbar';
export * from './ScrollTop/ScrollTop';
