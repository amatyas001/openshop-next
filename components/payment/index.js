export * from './PaymentCancelled/PaymentCancelled';
export * from './PaymentConfirm/PaymentConfirm';
export * from './PaymentConfirmCard/PaymentConfirmCard';
export * from './PaymentConfirmControls/PaymentConfirmControls';
export * from './PaymentConfirmDetails/PaymentConfirmDetails';
export * from './PaymentError/PaymentError';
export * from './PaymentForm/PaymentForm';
export * from './PaymentFormControls/PaymentFormControls';
export * from './PaymentProgress/PaymentProgress';
export * from './PaymentReview/PaymentReview';
export * from './PaymentSuccess/PaymentSuccess';
