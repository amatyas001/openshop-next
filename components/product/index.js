export * from './ProductCard/ProductCard';
export * from './ProductControls/ProductControls';
export * from './ProductControlsColor/ProductControlsColor';
export * from './ProductControlsAmount/ProductControlsAmount';
export * from './ProductControlsButton/ProductControlsButton';
export * from './ProductControlsSize/ProductControlsSize';
