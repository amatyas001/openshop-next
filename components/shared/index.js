export * from './Button/Button';
export * from './Card/Card';
export * from './Field/Field';
export * from './Rating/Rating';
export * from './ScrollTop/ScrollTop';
export * from './Navbar/Navbar';
export * from './Footer/Footer';
export * from './Spinner/Spinner';
