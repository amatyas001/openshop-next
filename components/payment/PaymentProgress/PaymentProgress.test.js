import { Provider } from 'react-redux';
import createStore from 'redux-mock-store';
import { create, act } from 'react-test-renderer';
import { PaymentProgress } from '@app/components';
import { INITIAL_STATE } from '@app/config';

let tree;
const mockStore = createStore([]);
const statuses = ['review', 'form', 'confirm', 'success', 'cancelled', 'error'];

describe('<PaymentProgress />', () => {
  it('should render with initial state', () => {
    expect.assertions(1);
    act(() => {
      tree = create(
        <Provider store={mockStore(INITIAL_STATE)}>
          <PaymentProgress />
        </Provider>
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  statuses.forEach((status) => {
    it(`should render properly on ${status} state`, () => {
      expect.assertions(1);
      const store = mockStore({ ...INITIAL_STATE, payment: { status } });
      act(() => {
        tree = create(
          <Provider store={store}>
            <PaymentProgress />
          </Provider>
        );
      });
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
