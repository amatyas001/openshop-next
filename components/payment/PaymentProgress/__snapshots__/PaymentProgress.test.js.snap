// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentProgress /> should render properly on cancelled state 1`] = `
<div
  className="css-12glmx8"
>
  <p
    className="css-18poma1"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
  >
    Review
  </p>
  <svg
    className="css-0"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
    />
  </svg>
  <p
    className="css-18poma1"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
  >
    Payment Details
  </p>
  <svg
    className="css-0"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
    />
  </svg>
  <p
    className="css-18poma1"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
  >
    Finished
  </p>
</div>
`;

exports[`<PaymentProgress /> should render properly on confirm state 1`] = `
<div
  className="css-12glmx8"
>
  <p
    className="css-18poma1"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
  >
    Review
  </p>
  <svg
    className="css-0"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
    />
  </svg>
  <p
    className="css-18poma1"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
  >
    Payment Details
  </p>
  <svg
    className="css-0"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
    />
  </svg>
  <p
    className="css-18poma1"
    style={Object {}}
  >
    Finished
  </p>
</div>
`;

exports[`<PaymentProgress /> should render properly on error state 1`] = `
<div
  className="css-12glmx8"
>
  <p
    className="css-18poma1"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
  >
    Review
  </p>
  <svg
    className="css-0"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
    />
  </svg>
  <p
    className="css-18poma1"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
  >
    Payment Details
  </p>
  <svg
    className="css-0"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
    />
  </svg>
  <p
    className="css-18poma1"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
  >
    Finished
  </p>
</div>
`;

exports[`<PaymentProgress /> should render properly on form state 1`] = `
<div
  className="css-12glmx8"
>
  <p
    className="css-18poma1"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
  >
    Review
  </p>
  <svg
    className="css-0"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
    />
  </svg>
  <p
    className="css-18poma1"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
  >
    Payment Details
  </p>
  <svg
    className="css-0"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
    />
  </svg>
  <p
    className="css-18poma1"
    style={Object {}}
  >
    Finished
  </p>
</div>
`;

exports[`<PaymentProgress /> should render properly on review state 1`] = `
<div
  className="css-12glmx8"
>
  <p
    className="css-18poma1"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
  >
    Review
  </p>
  <svg
    className="css-0"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
    />
  </svg>
  <p
    className="css-18poma1"
    style={Object {}}
  >
    Payment Details
  </p>
  <svg
    className="css-0"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
    />
  </svg>
  <p
    className="css-18poma1"
    style={Object {}}
  >
    Finished
  </p>
</div>
`;

exports[`<PaymentProgress /> should render properly on success state 1`] = `
<div
  className="css-12glmx8"
>
  <p
    className="css-18poma1"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
  >
    Review
  </p>
  <svg
    className="css-0"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
    />
  </svg>
  <p
    className="css-18poma1"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
  >
    Payment Details
  </p>
  <svg
    className="css-0"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
    />
  </svg>
  <p
    className="css-18poma1"
    style={
      Object {
        "color": "#44337A",
        "fontWeight": "bold",
      }
    }
  >
    Finished
  </p>
</div>
`;
